File: src/main/python/model/tokenizer.py
class Tokenizer:
    def __init__(self, oov_token=""):
        self.word_index = {}
        self.oov_token = oov_token
        self.word_index[oov_token] = 1

    def fit_on_texts(self, texts):
        for text in texts:
            for s in text:
                if s not in self.word_index.keys():
                    self.word_index[s] = len(self.word_index) + 1

    def texts_to_sequences(self, texts):
        sequences = []
        for text in texts:
            sequence = []
            for s in text:
                if s not in self.word_index.keys():
                    if self.oov_token != "":
                        sequence.append(self.word_index.get(self.oov_token))
                else:
                    sequence.append(self.word_index.get(s))
            sequences.append(sequence)
        return sequences

    def sequences_to_texts(self, sequences):
        texts = []
        for sequence in sequences:
            text = []
            for s in sequence:
                text.append(list(self.word_index.keys())[list(self.word_index.values()).index(s)])
            texts.append(text)
        return texts

File: src/main/java/flute/utils/file_processing/Tokenizer.java
package flute.utils.file_processing;

/**
 * Created by Minology on 10:16 CH
 */
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class Tokenizer {
    private Map<String, Integer> wordIndex;
    private String outOfVocabularyToken;

    public Tokenizer() {
        wordIndex = new HashMap<>();
    }

    public Tokenizer(String outOfVocabularyToken) {
        wordIndex = new HashMap<>();
        this.outOfVocabularyToken = outOfVocabularyToken;
        this.wordIndex.put(outOfVocabularyToken, 1);
    }

    public void fitOnTexts(String[] texts) {
        for (String text : texts) {
            if (!wordIndex.containsKey(text)) {
                wordIndex.put(text, wordIndex.size() + 1);
            }
        }
    }

    public ArrayList<ArrayList<Integer>> textsToSequences(ArrayList<ArrayList<String>> texts) {
        ArrayList<ArrayList<Integer>> sequences = new ArrayList<>();
        for (ArrayList<String> text: texts) {
            ArrayList<Integer> sequence = new ArrayList<>();
            for (String s : text) {
                if (!wordIndex.containsKey(s)) {
                    sequence.add(wordIndex.get(outOfVocabularyToken));
                } else {
                    sequence.add(wordIndex.get(s));
                }
            }
            sequences.add(sequence);
        }
        return sequences;
    }

    public Map<String, Integer> getWordIndex() {
        return wordIndex;
    }
}
